cmd/config/internal/commands/annotate.go

```diff
@@ -39,32 +44,68 @@ func AnnotateCommand(parent string) *cobra.Command {
 }
 
 type AnnotateRunner struct {
-	Command    *cobra.Command
-	Values     []string
-	Kind       string
-	Name       string
-	ApiVersion string
-	Namespace  string
-	Path       string
+	Command            *cobra.Command
+	Values             []string
+	Kind               string
+	Name               string
+	ApiVersion         string
+	Namespace          string
+	Path               string
+	RecurseSubPackages bool
 }
 
 func (r *AnnotateRunner) runE(c *cobra.Command, args []string) error {
 	var input []kio.Reader
 	var output []kio.Writer
+	openAPIFileName, err := ext.OpenAPIFileName()
```

Comment:
We should revisit this function. It is strange that it returns an error.

Reply:
The method uses a method which should be overridden by upstream. This check is to handle any errors in that case.

Reply:
But in this case the reason we need to account for an error here is that we are using a bit of a hack to get the name. We should revisit the `ext` package to see if we can separate defining the name of the krmfile from the path to the file in the root package.